x-pack/plugin/sql/qa/jdbc/security/build.gradle

```diff
@@ -1,29 +1,22 @@
 import org.elasticsearch.gradle.internal.test.RestIntegTestTask
 
+apply plugin: 'elasticsearch.internal-test-artifact'
+
 dependencies {
   testImplementation project(':x-pack:plugin:core')
 }
 
 Project mainProject = project
 
-configurations.create('testArtifacts')
-
-TaskProvider testJar = tasks.register("testJar", Jar) {
-  appendix 'test'
-  from sourceSets.test.output
-}
-
-artifacts {
-  testArtifacts testJar
-}
-
 subprojects {
   // Use tests from the root security qa project in subprojects
   configurations.create('testArtifacts')
```

Comment:
Shouldn't the plugin create this configuration?

Reply:
no. this is the consuming project. the plugin is just for providing the artifact. for consuming we only added the `testArtifact` convenience method see https://github.com/elastic/elasticsearch/blob/fd6d966e44170c2e621d0501c9a94b76f62038c7/build.gradle#L57-L74